ddcommon-ffi/src/metrics.rs

```diff
@@ -0,0 +1,76 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022-Present Datadog, Inc.
+
+use crate::{slice::AsBytes, slice::CharSlice};
+use ddcommon::metric::Metric;
+
+#[must_use]
+#[no_mangle]
+pub extern "C" fn ddog_Vec_Metric_new() -> crate::Vec<Metric> {
+    crate::Vec::default()
+}
+
+#[no_mangle]
+pub extern "C" fn ddog_Vec_Metric_drop(_: crate::Vec<Metric>) {}
+
+#[repr(C)]
+pub enum PushMetricResult {
+    Ok,
+    Err(crate::Vec<u8>),
+}
+
+#[no_mangle]
+pub extern "C" fn ddog_Vec_Metric_PushResult_drop(_: PushMetricResult) {}
+
+/// Creates a new Tag from the provided `key` and `value` by doing a utf8
+/// lossy conversion, and pushes into the `vec`. The strings `key` and `value`
+/// are cloned to avoid FFI lifetime issues.
+///
+/// # Safety
+/// The `vec` must be a valid reference.
+/// The CharSlices `key` and `value` must point to at least many bytes as their
+/// `.len` properties claim.
+#[no_mangle]
+pub unsafe extern "C" fn ddog_Vec_Metric_push(
+    vec: &mut crate::Vec<Metric>,
+    key: CharSlice,
+    value: f64,
+) -> PushMetricResult {
+    let name = key.to_utf8_lossy().into_owned();
+    match Metric::new(name, value) {
+        Ok(metric) => {
+            vec.push(metric);
+            PushMetricResult::Ok
+        }
+        Err(err) => PushMetricResult::Err(err.as_bytes().to_vec().into()),
+    }
+}
```

Comment:
I'm a bit hesitant to add this to ddcommon. It has the potential to become common, but it's also possible that (for instance) telemetry will want more out of a metric, and thus will need to create a different one anyway.

So I would suggest considering moving this into profiling for the time being :)